### PR TITLE
Added event due to the ticket:PDEEXP-1393

### PR DESCRIPTION
**Pull request by sukhrobbekodilov**
### 📦 Pull Request

[Provide a general summary of the pull request here.]

### ✅ Fixed Issues

- [List any fixed issues here like: Fixes #XXXX]

### 🚨 Test instructions

[Describe any additional context required to test the PR/feature/bug fix.]

### ⚠️ Don't forget to add a [semver](https://semver.org/) label! 
Please 🚨 **ONLY ADD ONE** 🚨 of the following labels, failing to do so may lead to adverse versioning of your changes when published:
- `patch`: Bug Fix?
- `minor`: New Feature?
- `major`: Breaking Change?
- `skip-release`: It's unnecessary to publish this change.

### Special Note
Please avoid adding any of the `Priority` labels as they conflict with the labels above ☝️

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @magic-ext/algorand@23.2.1-canary.770.10283407914.0
  npm install @magic-ext/aptos@11.2.1-canary.770.10283407914.0
  npm install @magic-ext/avalanche@23.2.1-canary.770.10283407914.0
  npm install @magic-ext/bitcoin@23.2.1-canary.770.10283407914.0
  npm install @magic-ext/conflux@21.2.1-canary.770.10283407914.0
  npm install @magic-ext/cosmos@23.2.1-canary.770.10283407914.0
  npm install @magic-ext/ed25519@19.2.1-canary.770.10283407914.0
  npm install @magic-ext/farcaster@0.2.2-canary.770.10283407914.0
  npm install @magic-ext/flow@23.2.1-canary.770.10283407914.0
  npm install @magic-ext/gdkms@11.2.1-canary.770.10283407914.0
  npm install @magic-ext/harmony@23.2.1-canary.770.10283407914.0
  npm install @magic-ext/icon@23.2.1-canary.770.10283407914.0
  npm install @magic-ext/near@23.2.1-canary.770.10283407914.0
  npm install @magic-ext/oauth@22.2.1-canary.770.10283407914.0
  npm install @magic-ext/oauth2@9.2.1-canary.770.10283407914.0
  npm install @magic-ext/oidc@11.2.1-canary.770.10283407914.0
  npm install @magic-ext/polkadot@23.2.1-canary.770.10283407914.0
  npm install @magic-ext/react-native-bare-oauth@25.2.1-canary.770.10283407914.0
  npm install @magic-ext/react-native-expo-oauth@25.2.1-canary.770.10283407914.0
  npm install @magic-ext/solana@25.3.1-canary.770.10283407914.0
  npm install @magic-ext/sui@0.3.1-canary.770.10283407914.0
  npm install @magic-ext/taquito@20.2.1-canary.770.10283407914.0
  npm install @magic-ext/terra@20.2.1-canary.770.10283407914.0
  npm install @magic-ext/tezos@23.2.1-canary.770.10283407914.0
  npm install @magic-ext/webauthn@22.2.1-canary.770.10283407914.0
  npm install @magic-ext/zilliqa@23.2.1-canary.770.10283407914.0
  npm install @magic-sdk/commons@24.2.1-canary.770.10283407914.0
  npm install @magic-sdk/pnp@22.2.1-canary.770.10283407914.0
  npm install @magic-sdk/provider@28.2.1-canary.770.10283407914.0
  npm install @magic-sdk/react-native-bare@29.2.1-canary.770.10283407914.0
  npm install @magic-sdk/react-native-expo@29.2.1-canary.770.10283407914.0
  npm install @magic-sdk/types@24.0.6-canary.770.10283407914.0
  npm install magic-sdk@28.2.1-canary.770.10283407914.0
  # or 
  yarn add @magic-ext/algorand@23.2.1-canary.770.10283407914.0
  yarn add @magic-ext/aptos@11.2.1-canary.770.10283407914.0
  yarn add @magic-ext/avalanche@23.2.1-canary.770.10283407914.0
  yarn add @magic-ext/bitcoin@23.2.1-canary.770.10283407914.0
  yarn add @magic-ext/conflux@21.2.1-canary.770.10283407914.0
  yarn add @magic-ext/cosmos@23.2.1-canary.770.10283407914.0
  yarn add @magic-ext/ed25519@19.2.1-canary.770.10283407914.0
  yarn add @magic-ext/farcaster@0.2.2-canary.770.10283407914.0
  yarn add @magic-ext/flow@23.2.1-canary.770.10283407914.0
  yarn add @magic-ext/gdkms@11.2.1-canary.770.10283407914.0
  yarn add @magic-ext/harmony@23.2.1-canary.770.10283407914.0
  yarn add @magic-ext/icon@23.2.1-canary.770.10283407914.0
  yarn add @magic-ext/near@23.2.1-canary.770.10283407914.0
  yarn add @magic-ext/oauth@22.2.1-canary.770.10283407914.0
  yarn add @magic-ext/oauth2@9.2.1-canary.770.10283407914.0
  yarn add @magic-ext/oidc@11.2.1-canary.770.10283407914.0
  yarn add @magic-ext/polkadot@23.2.1-canary.770.10283407914.0
  yarn add @magic-ext/react-native-bare-oauth@25.2.1-canary.770.10283407914.0
  yarn add @magic-ext/react-native-expo-oauth@25.2.1-canary.770.10283407914.0
  yarn add @magic-ext/solana@25.3.1-canary.770.10283407914.0
  yarn add @magic-ext/sui@0.3.1-canary.770.10283407914.0
  yarn add @magic-ext/taquito@20.2.1-canary.770.10283407914.0
  yarn add @magic-ext/terra@20.2.1-canary.770.10283407914.0
  yarn add @magic-ext/tezos@23.2.1-canary.770.10283407914.0
  yarn add @magic-ext/webauthn@22.2.1-canary.770.10283407914.0
  yarn add @magic-ext/zilliqa@23.2.1-canary.770.10283407914.0
  yarn add @magic-sdk/commons@24.2.1-canary.770.10283407914.0
  yarn add @magic-sdk/pnp@22.2.1-canary.770.10283407914.0
  yarn add @magic-sdk/provider@28.2.1-canary.770.10283407914.0
  yarn add @magic-sdk/react-native-bare@29.2.1-canary.770.10283407914.0
  yarn add @magic-sdk/react-native-expo@29.2.1-canary.770.10283407914.0
  yarn add @magic-sdk/types@24.0.6-canary.770.10283407914.0
  yarn add magic-sdk@28.2.1-canary.770.10283407914.0
  ```
</details>

### CODE REVIEW

File: packages/@magic-sdk/types/src/modules/auth-types.ts [hunk 1]
```diff
@@ -90,13 +90,15 @@ export enum LoginWithMagicLinkEventOnReceived {
...
   VerifyEmailOtp = 'verify-email-otp',
+  VerifyMFACode = 'verify-mfa-code',
   Cancel = 'cancel',
 }
 
 export enum LoginWithEmailOTPEventOnReceived {
   EmailOTPSent = 'email-otp-sent',
   InvalidEmailOtp = 'invalid-email-otp',
   ExpiredEmailOtp = 'expired-email-otp',
+  MfaSentHandle = 'mfa_sent_handle',
```
Comment (khamdam-magic) [inline packages/@magic-sdk/types/src/modules/auth-types.ts:101]: Let's use dash `-` instead of underscore `_` here.

**Comment by hcote** [comment]
Screenshot 2024-07-26 at 11 17 15 AM

Note: tests are failing because the coverage is not met - can you add tests for these changes?

**Comment by hcote** [comment]
Screenshot 2024-07-26 at 11 18 21 AM

Deepsource is also failing because of the usage of `as any` - you should be able to just change it to `as string`

**Comment by hcote** [comment]
Just want to confirm this has been tested locally? If so, once the above issues are addressed, looks good!

**Review by hcote — APPROVED** [review]

**Review by khamdam-magic — APPROVED** [review]

**Pull request merged**